### ISSUE TITLE
KeyError: 'python-3.3.0-pro0.tar.bz2'

### ISSUE THREAD

**Issue by astrofrog** [conda-forge/conda-smithy#494]
I tried to rerender the [glue-core-feedstock](https://github.com/conda-forge/glue-core-feedstock) feedstock with conda-smithy 2.2.2 and ran into the following error:

```
$ conda smithy rerender
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
Fetching package metadata .............
Traceback (most recent call last):
  File "/Users/tom/miniconda3/bin/conda-smithy", line 6, in <module>
    sys.exit(conda_smithy.cli.main())
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_smithy/cli.py", line 275, in main
    args.subcommand_func(args)
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_smithy/cli.py", line 186, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 709, in main
    render_circle(env, config, forge_dir)
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 77, in render_circle
    forge_config.get('channels', {}).get('sources', tuple())
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 628, in compute_build_matrix
    mtx = special_case_version_matrix(meta, index)
  File "/Users/tom/miniconda3/lib/python3.5/site-packages/conda_build_all/version_matrix.py", line 223, in special_case_version_matrix
    py_vn = minor_vn(index[python_pkg.fn]['version'])
KeyError: 'python-3.3.0-pro0.tar.bz2'
```

**Comment by jakirkham** [comment]
I'm not able to reproduce this error, @astrofrog. Do you get this error consistently or is it sporadic? Could you please also provide some more information like `conda list -n root`? Also what OS are you using? Have you change `conda-forge.yml`? What does `.condarc` contain? Thanks.

**Comment by patricksnape** [comment]
I'm seeing something similar to this trying to renrender the dlib recipe:

```
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
Fetching package metadata .............
Traceback (most recent call last):
  File "/Users/pts08/miniconda/bin/conda-smithy", line 6, in <module>
    sys.exit(conda_smithy.cli.main())
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 289, in main
    args.subcommand_func(args)
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 200, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 644, in main
    render_run_docker_build(env, config, forge_dir)
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 77, in render_run_docker_build
    forge_config.get('channels', {}).get('sources', tuple())
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 565, in compute_build_matrix
    mtx = special_case_version_matrix(meta, index)
  File "/Users/pts08/miniconda/lib/python2.7/site-packages/conda_build_all/version_matrix.py", line 196, in special_case_version_matrix
    np_vn = minor_vn(index[numpy_pkg.fn]['version'])
KeyError: u'numpy-1.5.1-py26_pro0.tar.bz2'
```

**Comment by tschoonj** [comment]
Very similar problem here on macOS:

````
$ conda-smithy rerender
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
Fetching package metadata .............
Traceback (most recent call last):
  File "/Users/awf63395/anaconda2/bin/conda-smithy", line 6, in <module>
    sys.exit(conda_smithy.cli.main())
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_smithy/cli.py", line 271, in main
    args.subcommand_func(args)
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_smithy/cli.py", line 182, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 705, in main
    render_circle(env, config, forge_dir)
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 77, in render_circle
    forge_config.get('channels', {}).get('sources', tuple())
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 626, in compute_build_matrix
    mtx = special_case_version_matrix(meta, index)
  File "/Users/awf63395/anaconda2/lib/python2.7/site-packages/conda_build_all/version_matrix.py", line 196, in special_case_version_matrix
    np_vn = minor_vn(index[numpy_pkg.fn]['version'])
KeyError: u'numpy-1.5.1-py26_pro0.tar.bz2'
````

**Comment by astrofrog** [comment]
It seems this was a sporadic issue for me, as it's working now: https://github.com/conda-forge/glue-core-feedstock/pull/3

**Comment by patricksnape** [comment]
I had that error when my conda-smithy was out of date. I was also running Python 2.7 - try using Python 3?

**Comment by jcb91** [comment]
I've just encountered the same issue while attempting to rerender [jcb91/jupyter_highlight_selected_word-feedstock](https://github.com/jcb91/jupyter_highlight_selected_word-feedstock):

```
$ conda smithy rerender
Fetching package metadata ...........
Traceback (most recent call last):
  File "/home/josh/miniconda3/bin/conda-smithy", line 6, in <module>
    sys.exit(conda_smithy.cli.main())
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_smithy/cli.py", line 275, in main
    args.subcommand_func(args)
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_smithy/cli.py", line 186, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 709, in main
    render_circle(env, config, forge_dir)
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 77, in render_circle
    forge_config.get('channels', {}).get('sources', tuple())
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_smithy/configure_feedstock.py", line 628, in compute_build_matrix
    mtx = special_case_version_matrix(meta, index)
  File "/home/josh/miniconda3/lib/python3.5/site-packages/conda_build_all/version_matrix.py", line 223, in special_case_version_matrix
    py_vn = minor_vn(index[python_pkg.fn]['version'])
KeyError: 'python-3.3.0-pro0.tar.bz2'
```

This is using conda-smithy 2.2.2, on Ubuntu 16.04 LTS. Re-running the command doesn't seem to fix anything, so it's not sporadic in that sense. here's a root-env list:

```
$ conda list -n root
# packages in environment at /home/josh/miniconda3:
#
anaconda-client           1.6.2                    py35_0  
beautifulsoup4            4.5.3                    py35_0  
cffi                      1.9.1                    py35_0  
chardet                   2.3.0                    py35_0  
clyent                    1.2.2                    py35_0  
conda                     4.3.15                   py35_0  
conda-build               2.1.7                    py35_0  
conda-build-all           1.0.3                    py35_0    conda-forge
conda-env                 2.6.0                         0  
conda-smithy              2.2.2                    py35_0    conda-forge
conda-verify              2.0.0                    py35_0  
cryptography              1.7.1                    py35_0  
filelock                  2.0.7                    py35_0  
flake8                    2.5.1                    py35_0  
gitdb2                    2.0.0                    py35_0  
gitpython                 2.1.1                    py35_0  
idna                      2.2                      py35_0  
jinja2                    2.9.5                    py35_0  
libffi                    3.2.1                         1  
markupsafe                0.23                     py35_2  
mccabe                    0.3.1                    py35_0  
mock                      2.0.0                    py35_0  
openssl                   1.0.2j                        0  
patchelf                  0.9                           0  
pbr                       1.10.0                   py35_0  
pep8                      1.7.0                    py35_0  
pip                       8.1.2                    py35_0  
pkginfo                   1.4.1                    py35_0  
pyasn1                    0.1.9                    py35_0  
pycosat                   0.6.1                    py35_1  
pycparser                 2.17                     py35_0  
pycrypto                  2.6.1                    py35_4  
pyflakes                  1.5.0                    py35_0  
pygithub                  1.29                     py35_0    conda-forge
pyopenssl                 16.2.0                   py35_0  
python                    3.5.2                         0  
python-dateutil           2.6.0                    py35_0  
pytz                      2016.10                  py35_0  
pyyaml                    3.12                     py35_0  
readline                  6.2                           2  
requests                  2.12.4                   py35_0  
ruamel.yaml               0.13.14                  py35_0    conda-forge
ruamel_yaml               0.11.14                  py35_0  
setuptools                27.2.0                   py35_0  
six                       1.10.0                   py35_0  
smmap2                    2.0.1                    py35_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
wheel                     0.29.0                   py35_0  
xz                        5.2.2                         0  
yaml                      0.1.6                         0  
zlib                      1.2.8                         3
```

and config file contents:

```
$ cat ~/.condarc
channels:
  - defaults
  - conda-forge
```

while `conda-forge.yml` remains un-edited.

**Comment by jschueller** [comment]
same here

**Comment by jakirkham** [comment]
It looks like you are using `conda` 4.3.x, @jcb91. We do not currently support `conda` 4.3 only 4.2. Are you able to reproduce this with `conda` 4.2.x? If not, I'm inclined to close this as invalid and we can revisit once `conda` 4.3 is a reasonable target for `conda-smithy` support.

While it would be nice to support `conda` 4.3, that is impossible without getting that support in `conda-build-all`. Work on that is occurring in PR ( https://github.com/SciTools/conda-build-all/pull/83 ).

**Comment by jcb91** [comment]
> Are you able to reproduce this with conda 4.2.x?

No, using 4.2.16 seems to fix it for me, so that might well be it.

> We do not currently support conda 4.3 only 4.2

It seems then like it might be a good idea to temporarily alter the conda-smithy requirements to exclude conda 4.3?!

**Comment by jakirkham** [comment]
Yeah, it's a tricky situation. If we do that the `conda` solver won't actually obey us, instead it will pick an older version of `conda-smithy` that does not have a version constraint and then still pick the newest `conda`. The net effect is people still have this bug. 😒

**Comment by jcb91** [comment]
Ah, I see. Is there any way to just get conda-smithy to check the conda version, then print a more informative error message?

**Comment by jakirkham** [comment]
That seems like a good idea. Should be simple enough. Would you like to do a small PR?

**Comment by jcb91** [comment]
Sure, I'll give it a try!

**Comment by jakirkham** [comment]
Thanks @jcb91.

**Comment by jakirkham** [comment]
This is fixed by `conda-build-all` 1.0.4 (released from `conda-forge`'s fork). Based on initial testing `conda` 4.3 appears to just work with `conda-smithy` once the `conda` check is relaxed. Will do a patch release for `conda-smithy` once testing is complete.
